inference-engine/tests/functional/plugin/shared/src/subgraph_tests/softsign.cpp

```diff
@@ -0,0 +1,132 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <ie_core.hpp>
+#include <memory>
+#include <string>
+#include <tuple>
+#include <vector>
+#include <ie_plugin_config.hpp>
+
+#include "common_test_utils/common_utils.hpp"
+#include "functional_test_utils/blob_utils.hpp"
+#include "functional_test_utils/layer_test_utils.hpp"
+#include "functional_test_utils/plugin_cache.hpp"
+#include "ngraph_functions/pass/convert_prc.hpp"
+#include <legacy/ngraph_ops/power.hpp>
+#include "subgraph_tests/softsign.hpp"
+
+#include "ngraph_functions/builders.hpp"
+
+namespace LayerTestsDefinitions {
+
+std::string SoftsignTest::getTestCaseName(testing::TestParamInfo<softsignParams> obj) {
+    InferenceEngine::Precision netPrecision;
+    std::vector<size_t> inputShape;
+    std::string targetDevice;
+    std::map<std::string, std::string> configuration;
+    std::tie(netPrecision, targetDevice, configuration, inputShape) = obj.param;
+
+    std::ostringstream result;
+    result << "IS=" << CommonTestUtils::vec2str(inputShape) << "_";
+    result << "netPRC=" << netPrecision.name() << "_";
+    result << "targetDevice=" << targetDevice;
+    for (auto const& configItem : configuration) {
+        result << "_configItem=" << configItem.first << "_" << configItem.second;
+    }
+    return result.str();
+}
+
+void SoftsignTest::SetUp() {
+    InferenceEngine::Precision netPrecision;
+    std::map<std::string, std::string> tempConfig;
+    std::vector<size_t> inputShape;
+    std::tie(netPrecision, targetDevice, tempConfig, inputShape) = this->GetParam();
+    configuration.insert(tempConfig.begin(), tempConfig.end());
+
+    auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
+
+    auto params = ngraph::builder::makeParams(ngPrc, { inputShape });
+
+    auto abs = std::make_shared<ngraph::op::Abs>(params[0]);
+    auto add = std::make_shared<ngraph::op::PowerIE>(abs, 1, 1, 1);
+    auto power = std::make_shared<ngraph::op::PowerIE>(add, -1, 1, 0);
+    auto mul = std::make_shared<ngraph::op::Multiply>(power, params[0]);
+    ngraph::ResultVector results{ std::make_shared<ngraph::op::Result>(mul) };
+    function = std::make_shared<ngraph::Function>(results, params, "SoftSignTest");
+}
+
+void SoftsignTest::Run() {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+
+    LoadNetwork();
+    Infer();
+
+    const auto& actualOutputs = GetOutputs();
+    auto referenceOutputs = CalculateRefs();
+
+    Compare(referenceOutputs, actualOutputs);
+}
+
+std::vector<std::vector<std::uint8_t>> SoftsignTest::CalculateRefs() {
+    InferenceEngine::Precision netPrecision = std::get<0>(this->GetParam());
```

Comment:
Why is special reference function required?
If there is issue with precision, one of options is to modify threshold.

Reply:
There is no such operation as SoftSign in ngraph, so it is unable to check with that operation. There're also two Power operations: one in ngraph which is not used in inference engine conversion and one called PowerIE which is used there but is implemented in inference engine not ngraph, so ngraph interpreter doesn't recognize it. So I can do two things to implement the test: the one I did which is comparing the output only on GNA to see whether both patterns produce same results or implement the model with Power from ngraph and use this one in interpreter to compare model run on GNA with PowerIE and the one with Power run on ngraph interpreter.

Reply:
why not simply switch "function" variable after infer and before validate?